openms-wiki/contributor-faq.md

```diff
@@ -0,0 +1,269 @@
+
+## General
+
+The following section provides general information to new contributors.
+
+### I am new to OpenMS. What should I do first?
+
+* Check out the development version of OpenMS (see website).
+* Try and build OpenMS according to the installation instructions.
+* Read the OpenMS Coding Convention.
+* Read the OpenMS Tutorial.
+* Create a GitHub account
+* Register to the open-ms-general and open-ms-developers mailing list. (You can see the developers list only if you are logged in to Sourceforge and if you are a OpenMS developer).
+
+### What is the difference between an OpenMS tool and util?
+
+A tool starts its lifecycle in `UTILS` and may exist without being thoroughly tested. Tools may be promoted from `UTILS` to `TOOLS` if they are stable enough, are fully tested, fully documented, and a test workflow exists.
+
+### I have written a class for OpenMS. What should I do?
+
+Follow the [OpenMS coding conventions](coding-conventions.md).
+
+Coding style (brackets, variable names, etc.) must conform to the conventions.
+
+* The class and all the members must be documented thoroughly.
+* You can check your code with the tool  `tools/checker.php`. Call `php tools/checker.php` for detailed instructions.
+
+Please open a pull request and follow the [checklist](pull-request-checklist.md).
+
+## Troubleshooting
+
+The following section provides information about how to troubleshoot common OpenMS issues.
+
+### OpenMS complains about boost not being found but I'm sure its there!
+
+`CMake` got confused. Set up a new build directory and try again. If you build in-source (not recommended), deleting the `CMakeCache.txt` and `cmake` directory might help.
+
+## Build System
+
+The following questions are related to the build system.
+
+### What is CMake?
+
+`CMake` builds BuildSystems for different platforms, e.g. VisualStudio Solutions on Windows, Makefiles on Linux etc.
+This allows us to define in one central location (namely `CMakeLists.txt`) how OpenMS is build and have the platform specific stuff handled by `CMake`.
+View the [cmake website[(http://www.cmake.org) for more information.
+
+### How do I use CMake?
+
+See Installation instructions for your platform.
+In general, you call `CMake(.exe)` with some parameters to create the native build-system.
+Afterwards you can (but usually don't have to edit the current configuration using a GUI named `ccmake` (or `CMake-GUI` in Windows), which ships with `CMake`).
+
+> **_NOTE:_**  whenever `ccmake` is mentioned in this document, substitute this by `CMake-GUI` if your OS is Windows. You can also edit the `CMakeCache.txt` file directly.
+
+### How do I generate a build-system for Eclipse, KDevelop, CodeBlocks etc?
+
+Type `cmake` into a console. This will list the available code generators available on your platform, which you can pass to `CMake` using the `-G` option.
+
+### How do I switch to debug or release configuration?
```

Comment:
I don't think you have adapted what has been discussed in discord?

Reply:
- How do I update the openms.de?
- What are user definable CMake cache variables?
- I changed the contrib path, but re-running CMake won't change the library paths?
- What are the most useful make targets?
- CMake can't seem to find a Qt library (usually QtCore). What now?
- (Windows) What version of Visual Studio should I use?
- Class/Unit tests and TOPP/Tool tests
- How do I add a test for a new class?
- How do I add a new GUI test (for QT Gui classes) for the class MyClass?
- (Linux) All tests fail when you execute make test
- Release
- How can I find out which shared libraries are used by an application?
- How can I get a list of the symbols defined in a (shared) library or object file?
- How can I profile my code?
- (Linux) How do I check my code for memory leaks?

Reply:
I'll remove these questions

Reply:
done